marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/flow/flowRunner.sjs

```diff
@@ -132,6 +132,7 @@ function processContentWithStep(stepExecutionContext, contentArray, databaseWrit
   }
 
   invokeBeforeMain(stepExecutionContext, contentArray);
+  applyTargetCollectionsAdditivity(stepExecutionContext, contentArray);
```

Comment:
This needs to happen after the step main function is run, as we want to apply the logic to the output content array, not the input content array.